src/WebJobs.Extensions.DurableTask/ContextInterfaces/IDeterministicExecutionContext.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// Functionality available in all deterministic execution contexts, such as orchestrations or actor operations.
+    /// </summary>
+    public interface IDeterministicExecutionContext
+    {
+        /// <summary>
+        /// Gets the current date/time in a way that is safe for use in orchestrations and actor operations.
+        /// </summary>
+        /// <remarks>
+        /// This date/time value is derived from the orchestration or actor history. It always returns the same value
+        /// at specific points in the orchestrator function code, making it deterministic and safe for replay.
+        /// </remarks>
+        /// <value>The orchestration or actor's current date/time in UTC.</value>
+        DateTime CurrentUtcDateTime { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether the orchestration or operation is currently replaying itself.
+        /// </summary>
+        /// <remarks>
+        /// This property is useful when there is logic that needs to run only when *not* replaying. For example, certain types of application logging may become too noisy when duplicated
+        /// as part of replay. The application code could check to see whether the function is
+        /// being replayed and then issue the log statements when this value is <c>false</c>.
+        /// </remarks>
+        /// <value>
+        /// <c>true</c> if the orchestration or operation is currently being replayed; otherwise <c>false</c>.
+        /// </value>
+        bool IsReplaying { get; }
+
+        /// <summary>
+        /// Determines whether the current context is locked, and if so, what locks are currently owned.
+        /// </summary>
+        /// <param name="ownedLocks">the collection of owned locks.</param>
+        /// <remarks>
+        /// Note that the collection of owned locks can be empty even if the context is locked. This happens
+        /// if an orchestration calls a suborchestration without lending any locks.
+        /// </remarks>
+        /// <returns>true if the context already holds some locks.</returns>
```

Comment:
`true` and `false` literals should be surrounded by `<c>` tags. See `IsReplaying` above for an example.